### PR TITLE
soc: esp32c3: add mcuboot support

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -190,7 +190,7 @@ manifest:
       revision: cfd050ff38a9d028dc211690b2ec35971128e45e
       path: modules/lib/open-amp
     - name: openthread
-      revision: b7aa7686d3f4e326bd25abc234b209159977c26d
+      revision: 41bc49da49736fbdfdfa231f7b2311f29dcc4979
```
Comment (edmont) [inline west.yml:193]: Hi, any reason why to downgrade OpenThread here?

Comment (sylvioalves) [reply west.yml:193]: @edmont, no reason at all. Missing rebase. Thanks for notice.

Comment (sylvioalves) [reply west.yml]: Fixed!

**Review by glaubermaroto — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]
Tested and works fine!

One comment though: would be nice to provide some documentation on how to build and flash MCUboot to successfully chain-load Zephyr application. https://github.com/mcu-tools/mcuboot/blob/main/docs/readme-espressif.md does not specify what should be the offset when flashing MCUboot. It is 0x1000 for ESP32 and 0x0 for ESP32-C3 if I remember correctly. Would be nice to have that specified somewhere.

**Comment by sylvioalves** [comment]
> Tested and works fine!
> 
> One comment though: would be nice to provide some documentation on how to build and flash MCUboot to successfully chain-load Zephyr application. https://github.com/mcu-tools/mcuboot/blob/main/docs/readme-espressif.md does not specify what should be the offset when flashing MCUboot. It is 0x1000 for ESP32 and 0x0 for ESP32-C3 if I remember correctly. Would be nice to have that specified somewhere.

Thanks for that observation. I'll submit a PR to add this information in the docs.

**Review by uLipe — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

**Comment by sylvioalves** [comment]
@nashif Did we miss the window to merge this for release 3.0?

**Pull request merged**